lib/signal-list-view.js

```diff
@@ -1,132 +1,124 @@
-"use babel";
+/* @flow */
 
-import { SelectListView } from "atom-space-pen-views";
+import SelectListView from "atom-select-list";
 import _ from "lodash";
 
 import WSKernel from "./ws-kernel";
 import kernelManager from "./kernel-manager";
 import { log } from "./utils";
 import store from "./store";
 
-// View to display a list of grammars to apply to the current editor.
-export default class SignalListView extends SelectListView {
-  initialize() {
-    super.initialize(...arguments);
-
+export type KernelCommand = {
+  command: string,
+  payload: ?Kernelspec
+};
+
+export default class SignalListView {
+  basicCommands: Array<Object>;
+  onConfirmed: ?(command: KernelCommand) => void;
+  panel: ?atom$Panel;
+  previouslyFocusedElement: ?HTMLElement;
+  selectListView: SelectListView;
+  switchCommands: Array<Object>;
+  wsKernelCommands: Array<Object>;
+
+  constructor() {
     this.basicCommands = [
-      {
-        name: "Interrupt",
-        value: "interrupt-kernel"
-      },
-      {
-        name: "Restart",
-        value: "restart-kernel"
-      },
-      {
-        name: "Shut Down",
-        value: "shutdown-kernel"
-      }
+      { name: "Interrupt", value: "interrupt-kernel" },
+      { name: "Restart", value: "restart-kernel" },
+      { name: "Shut Down", value: "shutdown-kernel" }
     ];
 
     this.wsKernelCommands = [
-      {
-        name: "Rename session for",
-        value: "rename-kernel"
-      },
-      {
-        name: "Disconnect from",
-        value: "disconnect-kernel"
-      }
+      { name: "Rename session for", value: "rename-kernel" },
+      { name: "Disconnect from", value: "disconnect-kernel" }
+    ];
+
+    this.switchCommands = [
+      { name: "Switch to", value: "switch-kernel", payload: null }
     ];
 
     this.onConfirmed = null;
-    this.list.addClass("mark-active");
-    this.setItems([]);
+    this.selectListView = new SelectListView({
+      itemsClassList: ["mark-active"],
+      items: [],
+      filterKeyForItem: item => item.name,
+      elementForItem: item => {
+        const element = document.createElement("li");
+        element.textContent = item.name;
+        return element;
+      },
+      didConfirmSelection: item => {
+        log("Selected command:", item);
+        if (this.onConfirmed) this.onConfirmed(item);
+        this.cancel();
+      },
+      didCancelSelection: () => this.cancel(),
+      emptyMessage: "No running kernels for this file type."
+    });
   }
 
-  toggle() {
-    if (this.panel) {
-      this.cancel();
-    } else if (atom.workspace.getActiveTextEditor()) {
-      this.editor = atom.workspace.getActiveTextEditor();
-      this.attach();
-    }
+  _mapCommands(commands: Array<Object>, kernelSpec: Kernelspec) {
+    return commands.map(command => ({
+      name: `${command.name} ${kernelSpec.display_name} kernel`,
+      command: command.value,
+      payload: command.hasOwnProperty("payload") ? kernelSpec : null
```

Comment:
I think we can be a bit more clear here if we do
```js
payload: (command.value === "switch-kernel") ? kernelSpec : null
```
and remove the payload property from [this line](https://github.com/nteract/hydrogen/pull/758/files#diff-070a3f52b3f9f0cbce35dbf6a46558acR38).